### PR TITLE
WIP - Create a Filename for the File to be Downloaded

### PR DESCRIPTION
**Pull request by MaaikeB**
File consists of page title, minimum year and maximum year.
This so that the file does not get the same name every time, and it is difficult to find files back.

Dependend on https://github.com/OpenBudget/open-budget-data-api/pull/10

DONT MERGE YET - there is an error when downloading the CSV file

### CODE REVIEW

File: src/app/services/budgetkey-item.ts [hunk 1]
```diff
@@ -23,9 +23,10 @@ export class BudgetKeyItemService {
       encodeURIComponent(query);
   }
 
-  getDownloadUrl(query: string, format: string, headers: string[]): string {
+  getDownloadUrl(query: string, format: string, headers: string[], fileName: string): string {
     return '//next.obudget.org/api/download?query=' + encodeURIComponent(query) +
       '&format=' + format +
+      '&filename=' + fileName +
```
Comment (akariv) [inline src/app/services/budgetkey-item.ts]: I think you should `encodeURIComponent` the file name here before embedding in the URL

File: src/app/components/item-data.ts [hunk 2]
```diff
@@ -125,17 +125,41 @@ export class ItemQuestionsComponent implements OnDestroy {
...
+      entityName = this.store.item.page_title;
+    }
+
+    // Create the question
+    let question = '';
+    for (let i = 0; i < this.store.currentQuestion.parsed.length; i++) {
+      question = question + this.store.currentQuestion.parsed[i].value;
+    }
+
+    const fileName = decodeURIComponent(entityName + '__' + question);
```
Comment (akariv) [inline src/app/components/item-data.ts]: `decodeURIComponent` makes sense when we're decoding something from a url - which is not the case here...

Comment (MaaikeB) [reply src/app/components/item-data.ts]: fixed it

**Comment by MaaikeB** [comment]
@akariv please review fixes

**Pull request merged**